[jit] Split out Function into its own file

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#21539 [jit] Split out Function into its own file**

Some preparatory movement to prepare for the CU ownership overhaul.
Changes are purely mechanical.

Differential Revision: [D15741979](https://our.internmc.facebook.com/intern/diff/D15741979)

suo committed:
[jit] Split out Function into its own file

Some preparatory movement to prepare for the CU ownership overhaul.
Changes are purely mechanical.

Krovatkin commented on aten/src/ATen/core/jit_type.h:
```diff
@@ -1093,7 +1093,7 @@ using FunctionTypePtr = std::shared_ptr<FunctionType>;
 struct CAFFE2_API FunctionType : public Type {
   static FunctionTypePtr create(
-      std::shared_ptr<torch::jit::script::Function> function) {
+      std::shared_ptr<torch::jit::Function> function) {
```
flatenning namespaces?

suo replied:
yes, thought I might as well start now :)

Krovatkin commented on torch/csrc/jit/function.h:
```diff
@@ -0,0 +1,135 @@
+  mutable std::unique_ptr<FunctionSchema> schema_;
+};
+} // namespace jit
+} // namespace torch
```
nitpick: newline

Krovatkin commented on torch/csrc/jit/function.cpp:
```diff
@@ -0,0 +1,30 @@
+  check_single_output();
+}
+} // namespace jit
+} // namespace torch
```
nitpick: newline

Krovatkin reviewed (approved):
:shipit:

suo committed:
Update on "[jit] Split out Function into its own file"

[jit] Split out Function into its own file

Some preparatory movement to prepare for the CU ownership overhaul.
Changes are purely mechanical.

gh-metadata: pytorch pytorch 21539 gh/suo/57/head

suo committed:
Update on "[jit] Split out Function into its own file"

[jit] Split out Function into its own file

Some preparatory movement to prepare for the CU ownership overhaul.
Changes are purely mechanical.

gh-metadata: pytorch pytorch 21539 gh/suo/57/head

suo committed:
Update on "[jit] Split out Function into its own file"

[jit] Split out Function into its own file

Some preparatory movement to prepare for the CU ownership overhaul.
Changes are purely mechanical.

gh-metadata: pytorch pytorch 21539 gh/suo/57/head
